### DISCUSSION TITLE
How to debug/test changes I make to the repo locally?

### DISCUSSION THREAD

**Discussion by Yajur-Grover** [callstack/react-native-builder-bob#588] [Q&A] [answered]
Quick question regarding debugging create-react-native-library - How can I test changes I make locally to the library code? I want to use the local source code with my changes to run the create-react-native-library template to debug some errors I am getting when running it with the 6/18 nightly react-native version, as it is blocking the current integration for RNW. The wiki says I should point it to the following executable: `bob/packages/create-react-native-library/bin/create-react-native-library` but I am getting the following error: 
```
Error: Cannot find module '../lib/index'
Require stack:
- C:\new-account-repos\react-native-builder-bob\packages\create-react-native-library\bin\create-react-native-library
    at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
    at Module._load (node:internal/modules/cjs/loader:922:27)
    at Module.require (node:internal/modules/cjs/loader:1143:19)
    at require (node:internal/modules/cjs/helpers:119:18)
    at Object.<anonymous> (C:\new-account-repos\react-native-builder-bob\packages\create-react-native-library\bin\create-react-native-library:4:1)
    at Module._compile (node:internal/modules/cjs/loader:1256:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:86:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'C:\\new-account-repos\\react-native-builder-bob\\packages\\create-react-native-library\\bin\\create-react-native-library'
  ]
}
```
Changing the path to '../src/index.tx', which is the correct path to `index.ts` in the repo did not work either - got an 'invalid import' error on line 1.

**Comment by satya164** [comment]
Did you run `yarn watch` to build the sources?

https://github.com/callstack/react-native-builder-bob/blob/main/README.md#development-workflow

**Comment by Yajur-Grover** [comment]
After running `yarn watch`, I am able to run the executable successfully, but some of the edits I have made to `index.ts` and logging statements that I have added to `generateExampleApp.ts` are not appearing on the CLI. I have modified [create-react-native-library.cmd](https://github.com/callstack/react-native-builder-bob/blob/5052841763c6b50ea474f272480dc86c3f466138/packages/create-react-native-library/bin/create-react-native-library.cmd#L3) with the following command: `npx --yes create-react-native-library@latest --slug testcli --description testcli --author-name "React-Native-Windows Bot" --author-email 53619745+rnbot@users.noreply.github.com --author-url http://example.com/ --repo-url http://example.com/ --languages java-objc --type module-new --react-native-version 0.75.0-nightly-20240618-5df5ed1a8 --example vanilla testcli`. 

The command runs the same as if I run it locally, but it doesn't reflect any of the changes I have made after running `yarn watch`.

**Reply by satya164** [reply, marked as answer]
You need to point to the local JS file after building, not run the npx version - which fetches it from npm.
